### PR TITLE
Fix payload generation when Logger scope does not contain formatted message

### PR DESCRIPTION
**Pull request by karolz-ms**
Fixes issue https://github.com/Azure/diagnostics-eventflow/issues/345

### CODE REVIEW

**Comment by karolz-ms** [comment]
@ksmithRenweb would appreciate your input too (if you have time)

File: test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/LoggerInputTests.cs [hunk 1]
```diff
@@ -358,11 +358,12 @@ public void LoggerIsUsableViaILogger()
                 {
                     var factory = new LoggerFactory();
                     factory.AddEventFlow(diagnosticPipeline);
-                    var logger = new Logger<LoggerInputTests>(factory);
+                    var logger = (ILogger) new Logger<LoggerInputTests>(factory);
```
Comment (xiaomi7732) [inline test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/LoggerInputTests.cs]: Why not request a clear type like
```csharp
ILogger logger = new Logger<LoggerInputTests>(factory);
```

File: src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs [hunk 2]
```diff
@@ -102,27 +106,39 @@ private LogLevel ToLogLevel(Extensions.Logging.LogLevel loggerLevel)
...
         {
+            // Special case the "{OriginalFormat}" data key that is used to preserve the message before formatting was applied
+            // This data is omitted from the payload and replaced by formatted message.
+            // See https://github.com/aspnet/Logging/blob/master/src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs
+            const string OriginalFormatKey = "{OriginalFormat}";
+
+            // If the data is not structured, it will be omitted
+            bool unformattedMessageOmitted = true;
+
             var structuredSrc = src as IEnumerable<KeyValuePair<string, object>>;
```
Comment (xiaomi7732) [inline src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs]: How about:
```csharp
if(src is IEnumerable<KeyValuePari<string,object> structuredSrc)
{
...
}
```

File: src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs [hunk 3]
```diff
@@ -102,27 +106,39 @@ private LogLevel ToLogLevel(Extensions.Logging.LogLevel loggerLevel)
...
             if (structuredSrc != null)
             {
+                unformattedMessageOmitted = false;
                 using (var srcEnumerator = structuredSrc.GetEnumerator())
```
Comment (xiaomi7732) [inline src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs]: Why not just use `foreach` on `IEnumerable<T1,T2>`?

File: test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/LoggerInputTests.cs (hunk 1 above)
Comment (karolz-ms) [reply test/Microsoft.Diagnostics.EventFlow.Inputs.Tests/LoggerInputTests.cs]: Makes sense, fixed

File: src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs (hunk 2 above)
Comment (karolz-ms) [reply src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs]: Nice, fixed

File: src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs (hunk 3 above)
Comment (karolz-ms) [reply src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs]: That was refactored from original code that needed to use the enumerator directly. Fixed now!

**Comment by karolz-ms** [comment]
Thanks for the review, @xiaomi7732 Much appreciated

**Comment by ksmithRenweb** [comment]
@karolz-ms I missed the post, but I've looked at the commits. They look good. They also confirm my guess to be pretty close to accurate as to the intent. 

Thanks for getting to this so fast. My PR in my groups sdk's hasn't even finished yet. So I can remove my hacks. We always prefer not to have to maintain such work arounds.

**Pull request merged**